Technical Specification

GuillaumeDespaux opened the pull request:
Technical Up to date for now. 
Talk about :
- Dijkstra Algorithms
- API REST
- Data Verification 
- Depth First Search
- Breadth First Search

GuillaumeDespaux committed:
Create technical_specification.md

Template for the technical specification.

GuillaumeDespaux committed:
Technical specification improvement

Added the convention name and usage for naming file/folder, code etc...
Changing the name of files and organization.

GuillaumeDespaux committed:
Update technical_specification.md

GuillaumeDespaux committed:
End of Algorithm + Start of API

GuillaumeDespaux committed:
API Modification

GuillaumeDespaux committed:
Merge branch 'main' into technical

GuillaumeDespaux committed:
Dijkstra completion + logical, table of content

GuillaumeDespaux committed:
Typos + div center, Index

GuillaumeDespaux committed:
API modification, + Rename convention

Petite-Brioche0 reviewed (changes requested):
- Glossary should be 15.
- line 75 : to use an REST API and reponse in JSON -> to use a REST API and respond in JSON
- line 83 : Provide answer through -> Provide responses through
- line 91 : folder that store all -> folder that stores all
- line 143 : with it's equivalent -> with its equivalent
- line 227 : Permit the print -> Allows printing
- line 247 : Before continuing reading -> Before continuing to read
- line 326 : ensure the data are correct -> ensure the data is correct
- line 937 : the link to HomeBrew is not correctly setted up

GuillaumeDespaux committed:
Issues https://github.com/algosup/2024-2025-project-3-quickest-path-team-2/pull/4#pullrequestreview-2569611636

Glossary should be 15.
- [x] line 75 : to use an REST API and reponse in JSON -> to use a REST API and respond in JSON
- [x] line 83 : Provide answer through -> Provide responses through
- [x] line 91 : folder that store all -> folder that stores all
- [x] line 143 : with it's equivalent -> with its equivalent
- [x] line 227 : Permit the print -> Allows printing
- [x] line 247 : Before continuing reading -> Before continuing to read
- [ ] line 326 : ensure the data are correct -> ensure the data is correct (data is plural and not singular)
- [x] line 937 : the link to HomeBrew is not correctly setted up
